src/main/java/com/dataartisans/App.java

```diff
@@ -11,16 +14,58 @@
 import org.apache.flink.streaming.runtime.streamrecord.StreamRecord;
 
 public class App {
+    private static final String INPUT_PATH_PARAM = "inputPath";
+    private static final String OUTPUT_PATH_PARAM = "outputPath";
+
     public static void main( String[] args ) throws Exception {
-        StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+        final ParameterTool params = ParameterTool.fromArgs(args);
+
+        final StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
         env.setStreamTimeCharacteristic(TimeCharacteristic.EventTime);
 
-        DataStream<Event> eventStream = env.addSource(new EventGenerator());
+        DataStream<Event> eventStream = getStreamFromSource(env, params);
+
         eventStream.transform("OrderOperator", eventStream.getType(), new OrderOperator<>());
 
+        setStreamSink(eventStream, params);
+
         env.execute("Run Stream smoother");
     }
 
+    private static DataStream<Event> getStreamFromSource(StreamExecutionEnvironment env, ParameterTool appConfig) {
+        if(appConfig.has(INPUT_PATH_PARAM)) {
+            return env.readTextFile(appConfig.get(INPUT_PATH_PARAM)).map(new JsonToEventMapper());
+        }
+        return env.addSource(new EventGenerator());
+    }
+    
+    private static void setStreamSink(DataStream<Event> stream, ParameterTool appConfig) {
+        if(appConfig.has(OUTPUT_PATH_PARAM)) {
+            stream.map(new EventToJsonMapper())
+                    .writeAsText(appConfig.get(OUTPUT_PATH_PARAM));
+        } else {
+            stream.print();
+        }
+    }
+
+    public static class JsonToEventMapper implements MapFunction<String, Event> {
+        private static final ObjectMapper jsonMapper = new ObjectMapper();
```

Comment:
As far as I can tell this instantiates `ObjectMapper` only once per App execution, which is desired (as opposed to once per element map).

[This post](http://apache-flink-mailing-list-archive.1008284.n3.nabble.com/Read-JSON-file-as-input-tp11400p11407.html), on the other hand, recommends:

> I recommend to extend the RichMapFunction and instantiate a parser in the open() method.